src/main/java/com/xceptance/xlt/tools/jenkins/XltTask.java

```diff
@@ -579,14 +579,7 @@ private void performPostTestSteps(final Run<?, ?> run, final Launcher launcher,
         // delete any temporary directory with local XLT
         try
         {
-            FilePath tempProjectFolder = getTemporaryXltProjectFolder(run, launcher);
-            tempProjectFolder.deleteRecursive();
-
-            FilePath tempFolder = getTemporaryXltBaseFolder(run, launcher);
-            if (tempFolder.exists() && (tempFolder.list() == null || tempFolder.list().isEmpty()))
-            {
-                tempFolder.delete();
-            }
+            getTemporaryXltBuildFolder(run, launcher).deleteRecursive();
```

Comment:
This will delete the temporary XLT folders of all steps (job might invoke our plugin multiple times) of this run. Shouldn't we change that to delete the temporary XLT folder of the current run and stepId only?
 ```suggestion
            getTemporaryXltFolder(run, launcher).deleteRecursive();
```

Reply:
Done. Furthermore, XLT copies are now managed in a flat directory structure to avoid leaving empty project directories behind.